Sources/Algorithms/Combinations.swift

```diff
@@ -78,6 +78,10 @@ public struct CombinationsSequence<Base: Collection> {
       binomial(n: n, k: $0)
     }.reduce(0, +)
   }
+
+  /// The total number of combinations.
+  @inlinable
+  public var underestimatedCount: Int { count }
```

Comment:
Doesn't this conflict with the complexity requirement of `Sequence.underestimatedCount`?

```swift
  /// - Complexity: O(1), except if the sequence also conforms to `Collection`.
  ///   In this case, see the documentation of `Collection.underestimatedCount`.
  var underestimatedCount: Int { get }
```

The requirement is relaxed for non-random access collections, so not a problem for #219.

Reply:
Hum, indeed.  What is the complexity of your `count`?  If it's log(N) IMO this is acceptable; complexity requirements are somewhat fungible (e.g. `first` on a lazy filtered collection is arguably not O(1)).  Otherwise, I suppose not.

Oh, but they totally broke the complexity bound of `underestimatedCount`.  It was O(n) before ea49459b7594.  What a debacle.

Reply:
Actually, that O(1) constraint is meaningless as outlined [here](https://github.com/apple/swift/issues/70755).  You should ignore it.